Client/src/GameContext/ClientEntityPool.cpp

```diff
@@ -2,31 +2,64 @@
 // Created by hippolyteb on 11/29/16.
 //
 
+#include <Messages/ReceivedNetworkPayloadMessage.hpp>
+#include <Json/Json.hpp>
 #include "ClientEntityPool.hpp"
 
 void ClientEntityPool::Draw(sf::RenderTexture &target, TextureBag &bag) {
     for (const auto& i : _pool)
     {
-        if (i->ImplementTrait(Trait::Drawable)) {
-            IDrawable *entity = dynamic_cast<IDrawable*>(i.GetInstance());
+        if (i.second->ImplementTrait(Trait::Drawable)) {
+            IDrawable *entity = dynamic_cast<IDrawable*>(i.second.GetInstance());
             auto renderTarget = entity->getRenderTexture();
-            if (renderTarget == nullptr)
-                renderTarget = entity->createRenderTexture((unsigned int) i->GetRenderRect().x, (unsigned int) i->GetRenderRect().y);
-            entity->Draw(renderTarget, bag);
-            renderTarget->display();
-            sf::Sprite sprite;
-            sprite.setTexture(renderTarget->getTexture());
-            sprite.setPosition(i->GetPosition().x, i->GetPosition().y);
-            target.draw(sprite);
+            auto sprite = entity->getSprite();
+            if (sprite == nullptr)
+                sprite = entity->createSprite();
+            if (renderTarget == nullptr || entity->NeedRedraw())
```

Comment:
We used to recreate a new Sprite at each frame. Now an entity can set a flag that inform the drawing system that it needs to be redrawn. Otherwise the same Sprite is used over and over again.
It's not a big performance improvement, but it helps.